expression/builtin_miscellaneous_vec_test.go

```diff
@@ -108,3 +112,103 @@ func BenchmarkVectorizedBuiltinMiscellaneousEvalOneVec(b *testing.B) {
 func BenchmarkVectorizedBuiltinMiscellaneousFunc(b *testing.B) {
 	benchmarkVectorizedBuiltinFunc(b, vecBuiltinMiscellaneousCases)
 }
+
+type counter struct {
+	count int
+}
+
+func (c *counter) add(diff int) int {
+	c.count += diff
+	return c.count
+}
+
+func (s *testEvaluatorSuite) TestSleepVectorized(c *C) {
+	ctx := mock.NewContext()
+	sessVars := ctx.GetSessionVars()
+
+	fc := funcs[ast.Sleep]
+	ft := eType2FieldType(types.ETReal)
+	col0 := &Column{RetType: ft, Index: 0}
+	f, err := fc.getFunction(ctx, []Expression{col0})
+	c.Assert(err, IsNil)
+	input := chunk.NewChunkWithCapacity([]*types.FieldType{ft}, 1024)
+	result := chunk.NewColumn(ft, 1024)
+	errCnt := counter{}
+	warnCnt := counter{}
+
+	// non-strict model
+	sessVars.StrictSQLMode = false
+	input.AppendFloat64(0, 1)
+	err = f.vecEvalInt(input, result)
+	c.Assert(err, IsNil)
+	c.Assert(result.GetInt64(0), Equals, int64(0))
+	c.Assert(sessVars.StmtCtx.WarningCount(), Equals, uint16(warnCnt.add(0)))
+
+	input.Reset()
+	input.AppendFloat64(0, -1)
+	err = f.vecEvalInt(input, result)
+	c.Assert(err, IsNil)
+	c.Assert(result.GetInt64(0), Equals, int64(0))
+	c.Assert(sessVars.StmtCtx.WarningCount(), Equals, uint16(warnCnt.add(1)))
+
+	input.Reset()
+	input.AppendNull(0)
+	err = f.vecEvalInt(input, result)
+	c.Assert(err, IsNil)
+	c.Assert(result.GetInt64(0), Equals, int64(0))
+	c.Assert(sessVars.StmtCtx.WarningCount(), Equals, uint16(warnCnt.add(1)))
+
+	input.Reset()
+	input.AppendNull(0)
+	input.AppendFloat64(0, 1)
+	input.AppendFloat64(0, -1)
+	err = f.vecEvalInt(input, result)
+	c.Assert(err, IsNil)
+	c.Assert(result.GetInt64(0), Equals, int64(0))
+	c.Assert(result.GetInt64(1), Equals, int64(0))
+	c.Assert(result.GetInt64(2), Equals, int64(0))
+	c.Assert(sessVars.StmtCtx.WarningCount(), Equals, uint16(warnCnt.add(2)))
+
+	// for error case under the strict model
+	sessVars.StrictSQLMode = true
+	input.Reset()
+	input.AppendNull(0)
+	err = f.vecEvalInt(input, result)
+	c.Assert(err, NotNil)
+	c.Assert(result.GetInt64(0), Equals, int64(0))
+	ec, _ := sessVars.StmtCtx.NumErrorWarnings()
+	c.Assert(ec, Equals, uint16(errCnt.add(1)))
+
+	sessVars.StmtCtx.SetWarnings(nil)
+	input.Reset()
+	input.AppendFloat64(0, -2.5)
+	err = f.vecEvalInt(input, result)
+	c.Assert(err, NotNil)
+	c.Assert(result.GetInt64(0), Equals, int64(0))
+	ec, _ = sessVars.StmtCtx.NumErrorWarnings()
+	c.Assert(ec, Equals, uint16(errCnt.add(1)))
+
+	//// strict model
+	input.Reset()
+	input.AppendFloat64(0, 0.5)
+	start := time.Now()
+	err = f.vecEvalInt(input, result)
+	c.Assert(err, IsNil)
+	c.Assert(result.GetInt64(0), Equals, int64(0))
+	sub := time.Since(start)
+	c.Assert(sub.Nanoseconds(), GreaterEqual, int64(0.5*1e9))
+
+	input.Reset()
+	input.AppendFloat64(0, 3)
```

Comment:
A tiny adjustment for this case is changing to multiple rows of input. Then we can check the results is `[0, 1, 1]`
```suggestion
	input.AppendFloat64(0, 0.01)
	input.AppendFloat64(0, 2)
	input.AppendFloat64(0, 1)
```